examples/github-deploy.yml

```diff
@@ -1,6 +1,11 @@
 name: Build and upload to PyPI
 
-on: [push, pull_request, release]
+on:
+  - push
+  - pull_request
+  - release:
```

Comment:
I'd comment the release lines out then add above it:

```python
# alternatively, to publish when a GitHub Release is created, use the following rule:
```

The `push` should already trigger on all tags, so that's fine for the current push-on-tag example; this is for the push-on-release example that's commented out below.

I'm a fan of push on release because it a) Forces a GitHub release, which GitHub allows users to "watch". You don't get a notification when just a tag gets pushed. It also b) Lets you build on tag, verify it works, and then release on the existing tag - if it fails, you don't get a bad PyPI release, and you can move a GH tag if you need to.

Reply:
Thank you @henryiii! Definitely hear you on this, I think deploy-on-release will be a great option for many projects. I just prefer build-on-tag as a more consistent experience across build providers.

Quick note on syntax- according to [this](https://github.community/t/array-of-configurable-trigger-events/16325) the contents of `on` is an object, not a list, so I -think- the syntax is

```yaml
on:
  push:
  pull_request:
  release:
    more_params: "something"
```

Reply:
Huh, but before, we had `on: [push, pull_request, release]`? That's why I turned it into a list :-/

But yes, I'll change it!

Reply:
I'm fine with the deploy-on-release; I was just explaining why I think it's worth adding this as an alternative as a comment to an example.

I think it allows a list of strings or a mapping, so the original was fine, and this is reworked one is fine, but the list of one-mapping-each was probably incorrect.

Reply:
Like this?

Reply:
What you have is fine, I would probably have done it like this:

```yaml
on:
  push:
  pull_request:
# alternatively, to publish when a (published) GitHub Release is created, add the following:
# release:
#   types:
#     - published
```

Reply:
Yeah, this keeps more similarity to `example/github-minimal.yml`. That's why I went for this.